gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/hive/UnpartitionedTableFileSetTest.java

```diff
@@ -44,11 +48,47 @@ public void testHiveTableLocationNotMatchException() throws Exception {
         Mockito.when(helper.getDataset()).thenReturn(hiveDataset);
         Mockito.when(helper.getExistingTargetTable()).thenReturn(Optional.of(existingTargetTable));
         Mockito.when(helper.getTargetTable()).thenReturn(table);
+        // Mock filesystem resolver
+        FileSystem mockFS = Mockito.mock(FileSystem.class);
+        Mockito.when(helper.getTargetFs()).thenReturn(mockFS);
+        Mockito.when(mockFS.resolvePath(Mockito.any())).then(returnsFirstArg());
+
+        Mockito.when(helper.getExistingEntityPolicy()).thenReturn(HiveCopyEntityHelper.ExistingEntityPolicy.ABORT);
+        MetricContext metricContext = MetricContext.builder("testUnpartitionedTableFileSet").build();
+        EventSubmitter eventSubmitter = new EventSubmitter.Builder(metricContext,"loc.nomatch.exp").build();
+        Mockito.when(helper.getEventSubmitter()).thenReturn(eventSubmitter);
+        UnpartitionedTableFileSet upts = new UnpartitionedTableFileSet("testLocationMatch",hiveDataset,helper);
+        List<CopyEntity> copyEntities = (List<CopyEntity>)upts.generateCopyEntities();
+    }
+
+    @Test
+    public void testHiveTableLocationMatchDifferentPathsResolved() throws Exception {
+        Path testPath = new Path("/testPath/db/table");
+        Path existingTablePath = new Path("/existing/testPath/db/table");
```

Comment:
Is it worth adding a test case comparing logical and physical paths to be explicit about what we're testing the difference between? If existing is logical and user specified is physical, then we don't want it to use the physical path right? 

Also why are the partitioned and snapshot cases different? For partitioned cases if paths differ we throw and error and abort, but for snapshot we can check the policy and overwrite location as needed.

Reply:
I think if the existing is logical and the user specified is physical, then the job should be marked as FAILED and thus never pass is what ended up being decided. I can add a test case against that but adding test cases is a bit moot since I'm already mocking the FileSystem class response when trying to resolve the paths together.

Snapshot tables tend to be overwritten much more often. E.g. When a user has a complete snapshot of some database collected on 01/03/2022 and is registered under a date path, then they ingest a new snapshot of their data on 01/06/2022. The most common behavior is to drop their old table and update hive to point to the newly ingested table.

For partitioned tables, tables should not be registered under a date partition, so the occurrence of table locations changing is much lower. Also if the table location is changed, then the user loses all their past partitions of data which is almost always unwanted. So in this scenario they probably should not be registering under the same table as before.

Reply:
Thanks, this is a helpful explanation. It sounds like this will work for our main area of interest, the snapshot tables. I agree it's a bit hard to test since you're mocking the FS response.